src/DGtal/kernel/domains/Linearizer.ih

```diff
@@ -0,0 +1,224 @@
+          {
+            aPoint[ linearizer_helper<TStorageOrder, N, 0>::dim ] = anIndex;
+          }
+      };
+
+  } // anonymous namespace
+
+  /// Linearized index of a point, given the domain lower-bound and extent.
+  template <typename TSpace, typename TStorageOrder>
+  typename Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::Size
+  Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::
+      getIndex( Point aPoint, Point const& aLowerBound, Extent const& anExtent )
+    {
+      aPoint -= aLowerBound;
+      return linearizer_impl<Size, TStorageOrder, Domain::dimension>::apply(aPoint, anExtent);
+    }
+
+  /// Linearized index of a point, given the domain extent.
+  template <typename TSpace, typename TStorageOrder>
+  typename Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::Size
+  Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::
+      getIndex( Point aPoint, Extent const& anExtent )
+    {
+      return linearizer_impl<Size, TStorageOrder, Domain::dimension>::apply(aPoint, anExtent);
+    }
+
+  /// Linearized index of a point, given a domain.
+  template <typename TSpace, typename TStorageOrder>
+  typename Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::Size
+  Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::
+      getIndex( Point aPoint, Domain const& aDomain )
+    {
+      return linearizer_impl<Size, TStorageOrder, Domain::dimension>::apply(aPoint - aDomain.lowerBound(), aDomain.upperBound()-aDomain.lowerBound()+Point::diagonal(1));
+    }
+
+  /// De-linearization of an index, given the domain lower-bound and extent.
+  template <typename TSpace, typename TStorageOrder>
+  typename Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::Point
+  Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::
+      getPoint( Size anIndex, Point const& aLowerBound, Extent const& anExtent )
+    {
+      Point point;
+      delinearizer_impl<TStorageOrder, Domain::dimension>::apply(point, anExtent, anIndex);
+      return point + aLowerBound;
+    }
+
+  /// De-linearization of an index, given the domain extent.
+  template <typename TSpace, typename TStorageOrder>
+  typename Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::Point
+  Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::
+      getPoint( Size anIndex, Extent const& anExtent )
+    {
+      Point point;
+      delinearizer_impl<TStorageOrder, Domain::dimension>::apply(point, anExtent, anIndex);
+      return point;
+    }
+
+  /// De-linearization of an index, given a domain.
+  template <typename TSpace, typename TStorageOrder>
+  typename Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::Point
+  Linearizer< HyperRectDomain<TSpace>, TStorageOrder >::
+      getPoint( Size anIndex, Domain const& aDomain )
+    {
+      Point point;
+      linearizer_impl<Size, TStorageOrder, Domain::dimension>::apply(point, aDomain.upperBound()-aDomain.lowerBound()+Point::diagonal(1), anIndex);
```

Comment:
It will probably improve the computations if used many times but it was meant to stay a static class (to be as simple as possible). I propose to add in the documentation that the methods that use the extent parameter are more efficient.

... otherwise, we could add an additional non-static class meant for "heavy use" and constructible from the domain or the extent&lowerBound. What do you think ?